app/controllers/api/users.php

```diff
@@ -415,6 +415,18 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             }
         }
 
+        if ($providerType === 'email') {
+            $validator = new Email();
+            if (!$validator->isValid($identifier)) {
+                throw new Exception(Exception::GENERAL_INVALID_EMAIL);
+            }
+        } elseif ($providerType === 'sms') {
+            $validator = new Phone();
+            if (!$validator->isValid($identifier)) {
+                throw new Exception(Exception::GENERAL_INVALID_PHONE);
+            }
+        }
```

Comment:
Let's use `switch`es instead for these checks, easier to add more types in future without it getting messy

Reply:
added case for push